main.go

```diff
@@ -97,29 +123,48 @@ func main() {
 		m, err := migrate.New("file://./migration", connStr)
 		if err != nil {
 			time.Sleep(2 * time.Second)
+
 			m, err = migrate.New("file://./migration", connStr)
 			if err != nil {
-				log.Print(fmt.Errorf("failed to migration init: %s", err.Error()))
+				log.Print(fmt.Errorf("failed to migration init: %w", err))
 				return
 			}
 		}
+
 		err = m.Up()
 		if err != nil {
-			log.Print(fmt.Errorf("failed to migrate up: %s", err.Error()))
+			log.Print(fmt.Errorf("failed to migrate up: %w", err))
 			return
 		}
 
 		db, err := database.NewDB(connStr)
 		if err != nil {
 			time.Sleep(2 * time.Second)
+
 			db, err = database.NewDB(connStr)
 			if err != nil {
-				// log.Fatalf("failed to initialize db: %s", err.Error())
-				log.Print(fmt.Errorf("failed to initialize db: %s", err.Error()))
+				log.Print(fmt.Errorf("failed to initialize db: %w", err))
 				return
 			}
 		}
-		handler := crud_handler.NewHandler(db)
-		handler.RunServer()
+
+		cache := cache.NewLruCache(10)
+		service := service.NewService(db, cache)
+		myhandler := handler.NewHandler(service)
+		myhandler.RunServer()
 	}
 }
+
+type MockCache struct {
```

Comment:
This is not mock - this is NopCache and must be placed in the `cache` package or in `cache/nop` subpackage